src/couch/src/couch_file.erl

```diff
@@ -215,12 +215,26 @@ truncate(Fd, Pos) ->
 sync(Filepath) when is_list(Filepath) ->
     case file:open(Filepath, [append, raw]) of
         {ok, Fd} ->
-            try ok = file:sync(Fd) after ok = file:close(Fd) end;
+            try
+                case file:sync(Fd) of
+                    ok ->
+                        ok;
+                    Else ->
```

Comment:
I'd prefer to match`{error, Reason}` and then `erlang:error({fsync_error, Reason}).` Should file:sync return something else, the crash from the function_clause is still an error.

Reply:
Done.